tests/src/ApiClientTest.php

```diff
@@ -4,104 +4,157 @@
 
 namespace OpenEuropa\Tests\CdtClient;
 
-use OpenEuropa\CdtClient\Contract\ApiClientInterface;
-use OpenEuropa\CdtClient\Contract\TokenAwareInterface;
-use OpenEuropa\CdtClient\Endpoint\IdentifierEndpoint;
-use OpenEuropa\CdtClient\Endpoint\MainEndpoint;
-use OpenEuropa\CdtClient\Endpoint\RequestsEndpoint;
-use OpenEuropa\CdtClient\Endpoint\StatusEndpoint;
-use OpenEuropa\CdtClient\Endpoint\TokenEndpoint;
-use OpenEuropa\CdtClient\Endpoint\ValidateEndpoint;
-use OpenEuropa\CdtClient\Http\Download;
+use GuzzleHttp\Psr7\Response;
+use OpenEuropa\CdtClient\Contract\ApiFactoryInterface;
+use OpenEuropa\CdtClient\Exception\ValidationErrorsException;
+use OpenEuropa\CdtClient\Model\Response\ReferenceData;
 use OpenEuropa\CdtClient\Model\Response\Token;
-use OpenEuropa\Tests\CdtClient\Traits\ClientTestTrait;
+use OpenEuropa\CdtClient\Model\Response\Translation;
+use OpenEuropa\Tests\CdtClient\Traits\ApiTestTrait;
+use OpenEuropa\Tests\CdtClient\Traits\RequestModelTestTrait;
 use PHPUnit\Framework\TestCase;
 
 /**
  * @coversDefaultClass \OpenEuropa\CdtClient\ApiClient
  */
 class ApiClientTest extends TestCase
```

Comment:
This was the Achilles heel of the previous version. I've changed it to cover all methods with simple tests. The asserts are not sophisticated, but sufficient to detect any runtime errors.